nodes/edge_detection.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+import rospy
+import sys
+
+import cozmo
+from cozmo.util import degrees
+import matplotlib.pyplot as plt
+
+from visualization_msgs.msg import Marker
+
+# Finds the corner coordinates of a 4 sided surface
+# Assumes Cozmo starts perpendicular to an edge
+# Cozmo will visit each edge once
+def perpendicular_detector(robot: cozmo.robot.Robot):
+    x_min = sys.maxsize
+    x_max = -sys.maxsize - 1
+    y_min = sys.maxsize
+    y_max = -sys.maxsize - 1
+    x = []
+    y = []
+
+    corners = []
+    robot.set_lift_height(1).wait_for_completed()
+
+    for angle in [180, 90, 180, 180]:
+        while not robot.is_cliff_detected:
+            robot.drive_wheels(100, 100, duration=0.25)
+        x_curr = robot.pose.position.x
+        y_curr = robot.pose.position.y
+        x.append(x_curr)
+        y.append(y_curr)
+
+        x_min = min(x_curr, x_min)
+        x_max = max(x_curr, x_max)
+        y_min = min(y_curr, y_min)
+        y_max = max(y_curr, y_max)
+        robot.drive_wheels(-100, -100, duration=1.5)
+        robot.turn_in_place(degrees(angle)).wait_for_completed()
+
+    plt.figure(0)
+    plt.scatter(x, y)
+    plt.plot(x[0:2], y[0:2], 'r')
+    plt.plot(x[2:4], y[2:4], 'r')
+
+    plt.scatter([x_min, x_min, x_max, x_max], [y_min, y_max, y_min, y_max])
+    plt.plot([x_min, x_max], [y_min, y_min], 'b')
+    plt.plot([x_min, x_max], [y_max, y_max], 'b')
+    plt.plot([x_min, x_min], [y_min, y_max], 'b')
+    plt.plot([x_max, x_max], [y_min, y_max], 'b')
+
+    plt.savefig('edge.png')
+    plt.show()
+
+# Finds the corner coordinates of a 4 sided surface
+# Cozmo will continuously find coordinates of the edges of the surface
+# until new coordinates are no longer expanding the known dimensions
+def generic_detector(robot: cozmo.robot.Robot):
```

Comment:
I don't understand the benefit of having this detector and the one above. If there's a clear benefit/difference you need to mention it